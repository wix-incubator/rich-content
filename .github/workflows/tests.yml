name: tests
on: push
env:
  captureVideos: 'true'
  captureImages: 'true'
  APPLITOOLS_BATCH_ID: ${{ github.sha }}

jobs:
  cleanup-runs:
    runs-on: ubuntu-latest
    steps:
      - uses: rokroskar/workflow-run-cleanup-action@v0.2.2
        if: "!startsWith(github.ref, 'refs/tags/') && github.ref != 'refs/heads/master'"
        env:
          GITHUB_TOKEN: '${{ secrets.GITHUB_TOKEN }}'

  e2e-visual:
    runs-on: ubuntu-18.04
    strategy:
      fail-fast: false
      matrix:
        ci_node_total: [5] # Set N number of parallel jobs you want to run tests on.
        ci_node_index: [0, 1, 2, 3, 4] # set N-1 indexes for parallel jobs

    steps:
      - uses: actions/checkout@v2
      - id: cache
        uses: actions/cache@master
        with:
          path: |
            node_modules
            */*/node_modules
            ~/.cache
          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
      - name: npm install + bootstrap
        if: steps.cache.outputs.cache-hit != 'true'
        run: npm ci
      - name: lerna link
        if: steps.cache.outputs.cache-hit == 'true'
        run: npm run link
      - name: build
        run: npm run build:test
      - name: Run http server with the app in the background
        run: |
          npm run e2e:app:run & npm run e2e:app:wait

      - name: Run tests with Knapsack Pro
        env:
          KNAPSACK_PRO_TEST_SUITE_TOKEN_CYPRESS: ${{ secrets.KNAPSACK_PRO_TEST_SUITE_TOKEN_CYPRESS }}
          KNAPSACK_PRO_CI_NODE_BUILD_ID: $(openssl rand -base64 32)
          KNAPSACK_PRO_CI_NODE_TOTAL: ${{ matrix.ci_node_total }}
          KNAPSACK_PRO_CI_NODE_INDEX: ${{ matrix.ci_node_index }}
          KNAPSACK_PRO_TEST_FILE_PATTERN: 'e2e/tests/*.e2e.js'
          APPLITOOLS_API_KEY: ${{ secrets.APPLITOOLS_API_KEY }}
          APPLITOOLS_SERVER_URL: ${{ secrets.APPLITOOLS_SERVER_URL }}
        run: |
          $(npm bin)/knapsack-pro-cypress
      - uses: actions/upload-artifact@v1
        if: failure()
        with:
          name: cypress-screenshots-${{ matrix.ci_node_index }}
          path: e2e/cypress/screenshots
      - uses: actions/upload-artifact@v1
        if: failure()
        with:
          name: cypress-videos-visual-${{ job.status }}-${{ matrix.ci_node_index }}
          path: e2e/cypress/videos

  e2e-visual-applitoolsSendCompletion:
    needs: e2e-visual
    if: always()
    runs-on: ubuntu-18.04
    steps:
      - name: Update Applitools batch status
        uses: wei/curl@v1.1.1
        with:
          args: -d "" -X POST https://wixeyesapi.applitools.com/api/externals/github/servers/github.com/commit/${{ env.APPLITOOLS_BATCH_ID }}/complete?apiKey=${{ secrets.APPLITOOLS_API_KEY }}

  content-state-tests:
    runs-on: ubuntu-18.04
    strategy:
      fail-fast: false
      matrix:
        ci_node_total: [3] # Set N number of parallel jobs you want to run tests on.
        ci_node_index: [0, 1, 2] # set N-1 indexes for parallel jobs

    steps:
      - uses: actions/checkout@v2
      - name: cache
        id: cache
        uses: actions/cache@master
        with:
          path: |
            node_modules
            */*/node_modules
            ~/.cache
          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
      - name: npm install + bootstrap
        if: steps.cache.outputs.cache-hit != 'true'
        run: npm ci
      - name: lerna link
        if: steps.cache.outputs.cache-hit == 'true'
        run: npm run link
      - name: build
        run: npm run build:test
      - name: Run http server with the app in the background
        run: npm run e2e:app:run & npm run e2e:app:wait

      - name: Run tests with Knapsack Pro
        env:
          KNAPSACK_PRO_TEST_SUITE_TOKEN_CYPRESS: ${{ secrets.KNAPSACK_PRO_TEST_SUITE_TOKEN_CYPRESS }}
          KNAPSACK_PRO_CI_NODE_BUILD_ID: $(openssl rand -base64 32)
          KNAPSACK_PRO_CI_NODE_TOTAL: ${{ matrix.ci_node_total }}
          KNAPSACK_PRO_CI_NODE_INDEX: ${{ matrix.ci_node_index }}
          KNAPSACK_PRO_TEST_FILE_PATTERN: 'e2e/tests/*.e2e.js'
          APPLITOOLS_IS_DISABLED: true
        run: |
          $(npm bin)/knapsack-pro-cypress --env MATCH_CONTENT_STATE=true
      - uses: actions/upload-artifact@v1
        if: failure()
        with:
          name: cypress-screenshots-${{ matrix.ci_node_index }}
          path: e2e/cypress/screenshots
      - uses: actions/upload-artifact@v1
        if: failure()
        with:
          name: cypress-videos-contentState-${{ job.status }}-${{ matrix.ci_node_index }}
          path: e2e/cypress/videos

  lint-and-unit-tests:
    runs-on: ubuntu-18.04
    steps:
      - uses: actions/checkout@v2
      - name: cache
        id: cache
        uses: actions/cache@master
        with:
          path: |
            node_modules
            */*/node_modules
            ~/.cache
          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
      - name: npm install + bootstrap
        if: steps.cache.outputs.cache-hit != 'true'
        run: npm ci
      - name: lerna link
        if: steps.cache.outputs.cache-hit == 'true'
        run: npm run link

      - run: npm run lint
      - run: npm run build
      - run: npm run test:unit

  notify-on-slack:
    runs-on: ubuntu-18.04
    needs: [e2e-visual, content-state-tests, lint-and-unit-tests]
    if: always()
    steps:
      - name: Get commit author
        id: commit-author
        run: |
          echo "::set-env name=commit-author::$(curl -s https://api.github.com/repos/wix-incubator/rich-content/commits/${{ github.sha }} | jq '.author.login' | sed 's/\"//g')"
      - name: Get PR number
        id: pr-number
        run: |
          echo "::set-env name=pr-number::$(curl -s https://api.github.com/repos/wix-incubator/rich-content/commits/${{ github.sha }} | jq '.commit.message' | sed 's/.*(#//g' | sed 's/[^0-9]//g')"
      - name: Get PR author
        id: pr-author
        run: |
          echo "::set-env name=pr-author::$(curl -s https://api.github.com/repos/wix-incubator/rich-content/pulls/${{ env.pr-number }} | jq '.merged_by.login' | sed 's/\"//g')"
      - name: Notify fail in Slack
        uses: voxmedia/github-action-slack-notify-build@v1.1.1
        if: github.ref == 'refs/heads/master' && (needs.e2e-visual.result == 'failure' || needs.content-state-tests.result == 'failure' || needs.lint-and-unit-tests.result == 'failure')
        with:
          channel_id: C0124DH7HHS #rich-content-ci
          status: FAILED ${{ env[env.commit-author] }} ${{ env[env.pr-author] }}
          color: danger

      # - name: Get current commit status
      #   id: current-status
      #   run: |
      #     echo "::set-output name=current-status::$(echo $(curl -s https://api.github.com/repos/wix-incubator/rich-content/commits/${{ github.sha }}/check-suites\?app_id\=15368 -H "Accept: application/vnd.github.antiope-preview+json" | jq '.check_suites | [.[] | select( .head_branch == "master") | .conclusion] | .[:2] | .[]' | sed 's/\"//g'))"
      - name: Get previous commit
        id: previous-commit
        run: |
          echo "::set-env name=previous-commit::$(curl -s https://api.github.com/repos/wix-incubator/rich-content/commits/${{ github.sha }} | jq '.parents[0].sha' | sed 's/\"//g')"
      # - name: Get previous commit status
      #   id: previous-status
      #   run: |
      #     echo "::set-output name=previous-status::$(echo $(curl -s https://api.github.com/repos/wix-incubator/rich-content/commits/${{ steps.previous-commit.outputs.previous-commit }}/check-suites\?app_id\=15368 -H "Accept: application/vnd.github.antiope-preview+json" | jq '.check_suites | [.[] | select( .head_branch == "master") | .conclusion] | .[:2] | .[]' | sed 's/\"//g'))"
      - name: Get previous commit status
        id: previous-status
        run: |
          echo "::set-env name=previous-status::$(echo $(curl -s https://api.github.com/repos/wix-incubator/rich-content/actions/workflows/tests.yml/runs\?branch\=master | jq '.workflow_runs | .[] | select( .head_sha == "${{ env.previous-commit }}") | .conclusion' | sed 's/\"//g'))"
      - name: Notify recovery in Slack
        uses: voxmedia/github-action-slack-notify-build@v1.1.1
        if: github.ref == 'refs/heads/master' && env.previous-status == 'failure' && (needs.e2e-visual.result == 'success' && needs.content-state-tests.result == 'success' && needs.lint-and-unit-tests.result == 'success')
        with:
          channel_id: C0124DH7HHS #rich-content-ci
          status: 'RECOVERED  :celebrate:'
          color: '#28a745'
    env:
      SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
      Barackos: <@UM3M904MU>
      barboaron: <@UMJ56Q30W>
      danarish4: <@UEBDPSU2W>
      lxgreen: <@U6FQCJ5GT>
      maimonav: <@UMUB0DSLD>
      mharsat: <@UJAAZU58B>
      noamtwix: <@UEBDQJBMG>
      oded-soffrin: <@U0MN7A3H8>
      ShamaySapir: <@UM48FG657>
      shaulgo: <@U0SDJ5S30>
      talevy17: <@UM24QHHAR>
      michael-wix: ' - Michael just added translations'
      wix-babel-ng: " - Michael's bot added new translations :pokesquirtlewalk:"
