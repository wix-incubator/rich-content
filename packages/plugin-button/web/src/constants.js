/* eslint-disable camelcase */
export const BUTTON_TYPE = 'wix-draft-plugin-button';
export const ALIGN_CENTER = 'center';
export const settingsTabValue = 'settings';
export const designTabValue = 'design';
export const BUTTON_TYPES = ['primary', 'secondary'];
export const DEFAULT_COLOR = '#000000';
export const DEFAULT_SELECTION_COLOR = '#000000';
export const COLOR_PICKER_TYPE = Object.freeze({
  TEXT_COLOR: 'textColor',
  BORDER_COLOR: 'borderColor',
  BACKGROUND_COLOR: 'backgroundColor',
});

export const COLORS = Object.freeze({
  color1: '#FFFFFF',
  color2: '#D5D4D4',
  color3: '#000000',
  color4: '#000000',
  color5: '#000000',
  color6: '#ABCAFF',
  color7: '#81B0FF',
  color8: '#0261FF',
  color9: '#0141AA',
  color10: '#012055',
});

export const DEFAULT_CONFIG = {
  alignment: ALIGN_CENTER,
  size: 'small',
  width: 'fit-content',
};

export const buttonPreviews = colors => [
  {
    className: 'button_primary',
    border: '0px solid #' + colors.color8,
    borderRadius: '0px',
    borderWidth: '0px',
    background: colors.color8,
    color: colors.color1,
    borderColor: colors.color8,
  },
  {
    className: 'button_secondary',
    border: '1px solid ' + colors.color8,
    borderRadius: '0px',
    borderWidth: '1px',
    background: colors.color1,
    color: colors.color8,
    borderColor: colors.color8,
  },
  {
    className: 'button_secondary',
    border: '1px solid ' + colors.color8,
    borderRadius: '0px',
    borderWidth: '1px',
    background: colors.color7,
    color: colors.color8,
    borderColor: colors.color8,
  },
  {
    className: 'button_secondary',
    border: '1px solid ' + colors.color8,
    borderRadius: '10px',
    borderWidth: '1px',
    background: colors.color7,
    color: colors.color8,
    borderColor: colors.color8,
  },
  {
    className: 'button_secondary',
    border: '1px solid ' + colors.color8,
    borderWidth: '5px',
    borderRadius: '0px',
    background: colors.color1,
    color: colors.color8,
    borderColor: colors.color8,
  },
];
