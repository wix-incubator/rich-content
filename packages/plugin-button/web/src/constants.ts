/* eslint-disable camelcase */
export const ALIGN_CENTER = 'center';
export const settingsTabValue = 'settings';
export const designTabValue = 'design';
export const BUTTON_TYPES = ['primary', 'secondary'];
export const DEFAULT_COLOR = '#000000';
export const DEFAULT_SELECTION_COLOR = '#000000';
export const COLOR_PICKER_TYPE = Object.freeze({
  TEXT_COLOR: 'textColor',
  BORDER_COLOR: 'borderColor',
  BACKGROUND_COLOR: 'backgroundColor',
});

export const DEFAULT_CONFIG = {
  alignment: ALIGN_CENTER,
  size: 'small',
  width: 'fit-content',
};
