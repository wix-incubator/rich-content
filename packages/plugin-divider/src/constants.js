export const DIVIDER_TYPE = 'wix-draft-plugin-divider';

export const LINE_SINGLE = 'single';
export const LINE_DOUBLE = 'double';
export const LINE_DASHED = 'dashed';
export const LINE_DOTTED = 'dotted';

export const LINE_TYPES = [LINE_SINGLE, LINE_DOUBLE, LINE_DASHED, LINE_DOTTED];

export const ALIGN_LEFT = 'left';
export const ALIGN_CENTER = 'center';
export const ALIGN_RIGHT = 'right';

export const SIZE_SMALL = 'small';
export const SIZE_MEDIUM = 'medium';
export const SIZE_LARGE = 'large';

export const DEFAULTS = {
  type: LINE_SINGLE,
  config: {
    dividerSize: SIZE_LARGE,
    dividerAlignment: ALIGN_CENTER
  }
};
