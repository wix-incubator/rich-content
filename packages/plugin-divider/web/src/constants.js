export const DIVIDER_TYPE = 'wix-draft-plugin-divider';

export const LINE_SINGLE = 'single';
export const LINE_DOUBLE = 'double';
export const LINE_DASHED = 'dashed';
export const LINE_DOTTED = 'dotted';

export const LINE_TYPES = Object.freeze([LINE_SINGLE, LINE_DOUBLE, LINE_DASHED, LINE_DOTTED]);

export const ALIGN_LEFT = 'left';
export const ALIGN_CENTER = 'center';
export const ALIGN_RIGHT = 'right';

export const SIZE_SMALL = 'small';
export const SIZE_MEDIUM = 'medium';
export const SIZE_LARGE = 'large';

export const SIZES = Object.freeze([SIZE_LARGE, SIZE_MEDIUM, SIZE_SMALL]);
export const ALIGNMENTS = [ALIGN_LEFT, ALIGN_CENTER, ALIGN_RIGHT];
export const NO_WRAP = 'nowrap';

export const DEFAULTS = Object.freeze({
  type: LINE_SINGLE,
  config: {
    size: SIZE_LARGE,
    alignment: ALIGN_CENTER,
    textWrap: NO_WRAP,
  },
});
