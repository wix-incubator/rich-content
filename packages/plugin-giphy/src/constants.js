export const PAGE_SIZE = 25;
export const WAIT_INTERVAL = 300;
export const SEARCH_TYPE = 'gifs';
export const MODAL_TYPE = 'flyOutModal';
export const GIPHY_TYPE = 'wix-draft-plugin-giphy';
export const GIPHY_API_KEY = 'wv9cEt7t1nu5wFYinAs73zwDzrrclGqk';

const gphApiClient = require('giphy-js-sdk-core');
export const giphyApiClient = gphApiClient(GIPHY_API_KEY);

export const DEFAULTS = {
  config: {
    size: 'content',
    alignment: 'center'
  },
};

export const MobileFullScreenCustomStyle = {
  overlay: {
    backgroundColor: 'transparent'
  },
  content: {
    top: 0,
    left: 0,
    overflow: 'hidden',
    paddingRight: '6px'
  }
};

export const DesktopFlyOutModalStyles = {
  overlay: {
    backgroundColor: 'rgba(0, 0, 0, 0)',
    zIndex: 10
  },
  content: {
    width: '265px',
    boxSizing: 'border-box',
    height: '357px',
    overflow: 'visible',
    border: '1px solid #ccc',
    paddingRight: '8px',
    paddingLeft: '18px',
    display: 'block',
    position: 'absolute',
    zIndex: 6,
    paddingTop: '8px'
  }
};
