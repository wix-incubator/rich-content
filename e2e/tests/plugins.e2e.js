/*global cy*/
import {
  PLUGIN_COMPONENT,
  PLUGIN_TOOLBAR_BUTTONS,
  DIVIDER_DROPDOWN_OPTIONS,
} from '../cypress/dataHooks';
import { DEFAULT_DESKTOP_BROWSERS } from '../tests/constants';

const eyesOpen = ({
  test: {
    parent: { title },
  },
}) =>
  cy.eyesOpen({
    appName: 'Plugins',
    testName: title,
    browser: DEFAULT_DESKTOP_BROWSERS,
  });

describe('plugins', () => {
  afterEach(() => cy.matchContentSnapshot());

  context('html', () => {
    before(function() {
      eyesOpen(this);
    });

    beforeEach('load editor', () => {
      cy.switchToDesktop();
    });

    after(() => cy.eyesClose());

    it('render html plugin toolbar', function() {
      cy.loadEditorAndViewer('empty').addHtml();
      cy.get(`[data-hook*=${PLUGIN_TOOLBAR_BUTTONS.EDIT}]`)
        .click({ multiple: true })
        .click();
      cy.eyesCheckWindow(this.test.title);
    });
  });

  context('divider', () => {
    before(function() {
      eyesOpen(this);
    });

    beforeEach('load editor', () => {
      cy.switchToDesktop();
    });

    after(() => cy.eyesClose());

    it('render plugin toolbar and change styling', () => {
      cy.loadEditorAndViewer('divider')
        .openPluginToolbar(PLUGIN_COMPONENT.DIVIDER)
        .openDropdownMenu();
      cy.eyesCheckWindow('render divider plugin toolbar');

      cy.clickToolbarButton(PLUGIN_TOOLBAR_BUTTONS.SMALL);
      cy.clickToolbarButton(PLUGIN_TOOLBAR_BUTTONS.ALIGN_LEFT);

      cy.get('.editor [data-hook=divider-double]')
        .parent()
        .click();
      cy.get('[data-hook*="PluginToolbar"]:first');

      cy.clickToolbarButton(PLUGIN_TOOLBAR_BUTTONS.MEDIUM);
      cy.clickToolbarButton(PLUGIN_TOOLBAR_BUTTONS.ALIGN_RIGHT);

      cy.get('.editor [data-hook=divider-dashed]')
        .parent()
        .click();
      cy.get('[data-hook*="PluginToolbar"]:first').openDropdownMenu(
        `[data-hook=${DIVIDER_DROPDOWN_OPTIONS.DOUBLE}]`
      );
      cy.eyesCheckWindow('change divider styling');
    });
  });

  context('map', () => {
    before('load editor', function() {
      eyesOpen(this);
      cy.switchToDesktop();
      cy.loadEditorAndViewer('map');
      cy.get('.dismissButton').eq(1);
    });

    after(() => cy.eyesClose());

    it('render map plugin toolbar and settings', () => {
      cy.openPluginToolbar(PLUGIN_COMPONENT.MAP);
      cy.eyesCheckWindow('render map plugin toolbar');
      cy.openMapSettings();
      cy.get('.gm-style-cc');
      cy.eyesCheckWindow('render map settings');
    });
  });

  context('file-upload', () => {
    before('load editor', function() {
      eyesOpen(this);
      cy.switchToDesktop();
      cy.loadEditorAndViewer('file-upload');
    });

    after(() => cy.eyesClose());

    it('render file-upload plugin toolbar', function() {
      cy.openPluginToolbar(PLUGIN_COMPONENT.FILE_UPLOAD);
      cy.eyesCheckWindow(this.test.title);
    });
  });

  context('drag and drop', () => {
    before('load editor', function() {
      eyesOpen(this);
      cy.switchToDesktop();
      cy.loadEditorAndViewer('dragAndDrop');
    });

    after(() => cy.eyesClose());

    // eslint-disable-next-line mocha/no-skipped-tests
    it.skip('drag and drop plugins', function() {
      cy.focusEditor();
      const src = `[data-hook=${PLUGIN_COMPONENT.IMAGE}] + [data-hook=componentOverlay]`;
      const dest = `span[data-offset-key="fjkhf-0-0"]`;
      cy.dragAndDropPlugin(src, dest);
      cy.get('img[style="opacity: 1;"]');
      cy.eyesCheckWindow(this.test.title);
    });
  });

  context('alignment', () => {
    before(function() {
      eyesOpen(this);
    });

    beforeEach('load editor', () => {
      cy.switchToDesktop();
    });

    after(() => cy.eyesClose());

    function testAtomicBlockAlignment(align) {
      it('align atomic block ' + align, function() {
        cy.loadEditorAndViewer('images').alignImage(align);
        cy.eyesCheckWindow(this.test.title);
      });
    }

    testAtomicBlockAlignment('left');
    testAtomicBlockAlignment('center');
    testAtomicBlockAlignment('right');
  });

  context('link preview', () => {
    before(function() {
      eyesOpen(this);
    });
    after(() => cy.eyesClose());

    beforeEach('load editor', () => cy.loadEditorAndViewer('link-preview'));

    it('change link preview settings', function() {
      cy.openPluginToolbar(PLUGIN_COMPONENT.LINK_PREVIEW);
      cy.setLinkSettings();
      cy.triggerLinkPreviewViewerUpdate();
      cy.eyesCheckWindow(this.test.title);
    });
    it('convert link preview to regular link', function() {
      cy.openPluginToolbar(PLUGIN_COMPONENT.LINK_PREVIEW);
      cy.clickToolbarButton('baseToolbarButton_replaceToLink');
      cy.triggerLinkPreviewViewerUpdate();
      cy.eyesCheckWindow(this.test.title);
    });
    it('backspace key should convert link preview to regular link', function() {
      cy.focusEditor()
        .type('{downarrow}{downarrow}')
        .type('{backspace}');
      cy.triggerLinkPreviewViewerUpdate();
      cy.eyesCheckWindow(this.test.title);
    });
    it('delete link preview', function() {
      cy.openPluginToolbar(PLUGIN_COMPONENT.LINK_PREVIEW).wait(100);
      cy.clickToolbarButton('blockButton_delete');
      cy.triggerLinkPreviewViewerUpdate();
      cy.eyesCheckWindow(this.test.title);
    });
  });

  context('convert link to preview', () => {
    before(function() {
      eyesOpen(this);
    });
    after(() => cy.eyesClose());
    beforeEach('load editor', () => cy.loadEditorAndViewer('empty'));

    it('should create link preview from link after enter key', function() {
      cy.insertLinkAndEnter('www.wix.com');
      cy.eyesCheckWindow(this.test.title);
    });

    it('should embed link that supports embed', function() {
      cy.insertLinkAndEnter('www.mockUrl.com');
      cy.eyesCheckWindow(this.test.title);
    });
  });

  context('list', () => {
    before(function() {
      eyesOpen(this);
    });

    beforeEach('load editor', () => cy.loadEditorAndViewer());

    after(() => cy.eyesClose());
    it('create nested lists using tab & shift-tab', function() {
      cy.loadEditorAndViewer()
        .enterParagraphs(['1. Hey I am an ordered list in depth 1.'])
        .tab()
        .enterParagraphs(['\n Hey I am an ordered list in depth 2.'])
        .tab()
        .enterParagraphs(['\n Hey I am an ordered list in depth 1.'])
        .tab({ shift: true })
        .enterParagraphs(['\n\n1. Hey I am an ordered list in depth 0.'])
        .enterParagraphs(['\n\n- Hey I am an unordered list in depth 1.'])
        .tab()
        .enterParagraphs(['\n Hey I am an unordered list in depth 2.'])
        .tab()
        .enterParagraphs(['\n Hey I am an unordered list in depth 1.'])
        .tab({ shift: true })
        .enterParagraphs(['\n\n- Hey I am an unordered list in depth 0.']);
      cy.eyesCheckWindow(this.test.title);
    });
  });
});
