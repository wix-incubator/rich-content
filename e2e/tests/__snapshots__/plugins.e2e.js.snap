exports[`plugins > divider > should render plugin toolbar #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 0,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 1,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 2,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 1,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 3,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 4,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 2,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 5,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 6,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 3,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 7,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 8,
      "text": "",
      "type": "unstyled"
    }
  ],
  "entityMap": {
    "0": {
      "data": {
        "config": {
          "alignment": "center",
          "size": "large",
          "textWrap": "nowrap"
        },
        "type": "single"
      },
      "mutability": "IMMUTABLE",
      "type": "wix-draft-plugin-divider"
    },
    "1": {
      "data": {
        "config": {
          "alignment": "center",
          "showDescription": true,
          "showTitle": true,
          "size": "content"
        },
        "type": "double"
      },
      "mutability": "IMMUTABLE",
      "type": "wix-draft-plugin-image"
    }
  }
};

exports[`plugins > divider > should render settings #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 0,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 1,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 2,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 1,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 3,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 4,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 0,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 5,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 6,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 1,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 7,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 8,
      "text": "",
      "type": "unstyled"
    }
  ],
  "entityMap": {
    "0": {
      "data": {
        "config": {
          "alignment": "center",
          "size": "large",
          "textWrap": "nowrap"
        },
        "type": "single"
      },
      "mutability": "IMMUTABLE",
      "type": "wix-draft-plugin-divider"
    },
    "1": {
      "data": {
        "config": {
          "alignment": "center",
          "showDescription": true,
          "showTitle": true,
          "size": "content"
        },
        "type": "double"
      },
      "mutability": "IMMUTABLE",
      "type": "wix-draft-plugin-image"
    }
  }
};

exports[`plugins > video > should render upload modal #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};

exports[`plugins > video > should enable to add a video from URI #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 0,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 1,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 2,
      "text": "",
      "type": "unstyled"
    }
  ],
  "entityMap": {
    "0": {
      "data": {
        "config": {
          "alignment": "center",
          "size": "content"
        },
        "isCustomVideo": false,
        "src": "https://www.youtube.com/watch?v=Xj3gU3jACe8"
      },
      "mutability": "IMMUTABLE",
      "type": "wix-draft-plugin-video"
    }
  }
};

exports[`plugins > video > should enable to add a custom video #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 0,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 1,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 2,
      "text": "",
      "type": "unstyled"
    }
  ],
  "entityMap": {
    "0": {
      "data": {
        "config": {
          "alignment": "center",
          "size": "content"
        },
        "isCustomVideo": true,
        "src": "http://mirrors.standaloneinstaller.com/video-sample/jellyfish-25-mbps-hd-hevc.mp4"
      },
      "mutability": "IMMUTABLE",
      "type": "wix-draft-plugin-video"
    }
  }
};

exports[`plugins > soundcloud > should render upload modal #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};

exports[`plugins > soundcloud > should enable to add a soundcloud URI #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 0,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 1,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 2,
      "text": "",
      "type": "unstyled"
    }
  ],
  "entityMap": {
    "0": {
      "data": {
        "config": {
          "alignment": "center",
          "size": "content"
        },
        "src": "https://soundcloud.com/nlechoppa/camelot"
      },
      "mutability": "IMMUTABLE",
      "type": "wix-draft-plugin-sound-cloud"
    }
  }
};

exports[`plugins > html > should render plugin toolbar #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};

exports[`plugins > html > should render settings #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "",
      "type": "unstyled"
    }
  ],
  "entityMap": {}
};
